### PR TITLE
Form Ruby on Rails

### PR DESCRIPTION
**Pull request by pastorp3**
The third project of Ruby on Rails, We create a model, controller, and views for User, Where you can create a new user and edit it and it will do the operation in the Data Base.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :users, only: %i[new create edit]
```
Comment (kobiyoyo) [inline config/routes.rb]: - [x]  Lets add `update` to the routes In config/routes.rb ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  def update
+    if @user.update(user_params)
+      flash.notice = "User '#{@user.username}' Updated!"
+      redirect_to edit_user_path
+    else
+      redirect_to edit_user_path
+      # flash.notice = "Some errors found try again!!!"
+      flash.notice = @user.errors.full_messages
+    end
```
Comment (kobiyoyo) [inline app/controllers/users_controller.rb]: This should `render edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  def edit; end
+
+  def update
+    if @user.update(user_params)
+      flash.notice = "User '#{@user.username}' Updated!"
+      redirect_to edit_user_path
+    else
+      redirect_to edit_user_path
+      # flash.notice = "Some errors found try again!!!"
```
Comment (kobiyoyo) [inline app/controllers/users_controller.rb]: Remove this commented code

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+class UsersController < ApplicationController
+  before_action :find_user, only: %i[edit update]
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (kobiyoyo) [inline app/controllers/users_controller.rb]: Remove this commented code

File: app/views/users/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+
+<%= form_for @user, url:  "/users/#{@user.id}", method: :patch do |f| %>
```
Comment (kobiyoyo) [inline app/views/users/edit.html.erb]: this should be written this way `<%= form_for @user do |u|%>`

File: app/views/users/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  <%= label_tag(:username) %>
+  <br>
+  <%= f.text_field :username, placeholder: 'Your name' %>
+  <br>
+  <br>
+  <%= label_tag(:email) %>
+  <br>
+  <%= f.text_field :email, placeholder: 'Your email'%>
```
Comment (kobiyoyo) [inline app/views/users/edit.html.erb]: `email_field` should be used here

File: app/views/users/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+
+<%= form_for @user, url:  "/users/#{@user.id}", method: :patch do |f| %>
+  <%= label_tag(:username) %>
```
Comment (kobiyoyo) [inline app/views/users/edit.html.erb]: This should be written like this `<%= u.label :username %>`

Note:implement changes to all the label tags

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [x] There should be a commented out html form ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- [x] Also  a commented out `form_tag` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- [x] Check comments below

# \[OPTIONAL\]

* [x] I encourage you to use a partial created for user form and reuse it in edit and new view
* [ ] validate the email format by using either a regular expression or a built-in URI method.### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by Ispirett — APPROVED** [review]
# Status Approved  :1st_place_medal: 
Great job folks have done well with all the changes made. :+1:

**Pull request merged**